lnwallet/chanfunding/canned_assembler.go

```diff
@@ -35,6 +35,11 @@ type ShimIntent struct {
 	// a normal channel. Until this height, it's considered frozen, so it
 	// can only be cooperatively closed by the responding party.
 	thawHeight uint32
+
+	// musig2 determines if the funding output should use musig2 to
+	// generate an aggregate key to use as the taproot-native multi-sig
+	// output.
+	musig2 bool
```

Comment:
How about a `legacy` bool, so that the current channel type becomes the exception and can be removed cleanly eventually?

Reply:
Don't really see a near future in us _removing_ the static key channel type, given that we still have tweakless in here all these years later as some older implementations still don't yet support it fully.